### PR TITLE
Case search validation Conditions

### PR DESCRIPTION
**Pull request by shubham1g5**
https://github.com/dimagi/commcare-core/pull/1138

Adds `error` to the query screen response from Formplayer to communicate any errors happening due to validations. 

````
{
  "displays": [
    {
      "text": "age",
      "id": "age",
       …,
      "error":  "Age must be greater than 18"
    }
  ],
  "type": "query"
}
````

### CODE REVIEW

File: src/test/resources/archives/caseclaim/suite.xml [hunk 1]
```diff
@@ -408,6 +307,18 @@
             <sort ref="id"/>
           </itemset>
         </prompt>
+        <prompt key="age">
```
Comment (shubham1g5) [inline src/test/resources/archives/caseclaim/suite.xml:310]: this is the only meaningful change in test file. Rest is - 
- earlier caseclaim folder was renamed to case_claim_post_in_query
- caseclaim.ccz was unarchived to caseclaim

Somehow github is screwing this diff.

**Review by snopoke — APPROVED** [review]

File: src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java [hunk 2]
```diff
@@ -483,7 +482,6 @@ private void answerQueryPrompts(FormplayerQueryScreen screen,
         if (queryDictionary != null) {
             screen.answerPrompts(queryDictionary);
         }
-        screen.refreshItemSetChoices();
```
Comment (snopoke) [inline src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java:486]: can the same call be removed above on line 433 in `handleQueryScreen`?

File: src/test/resources/archives/case_claim_post_in_entry/modules-1/forms-1.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			</instance>
+			<instance id="commcaresession" src="jr://instance/session"/>
+			<instance id="casedb" src="jr://instance/casedb"/>
+			<bind nodeset="/data/welcome_message"/>
+			<bind nodeset="/data/close_confirmation"/>
+			<bind nodeset="/data/close_reason" type="xsd:string" relevant="/data/close_confirmation = 'yes'"/>
+			<itext>
+				<translation lang="en" default="">
+					<text id="welcome_message-label">
+						<value>This is a case close form.  It lets you select a previously registered case and remove it from the phone. Its important to remove cases from the phone so that the user only sees the cases that they need to see.  However, all information collected about the case will still be stored on CommCareHQ.
```
Comment (snopoke) [inline src/test/resources/archives/case_claim_post_in_entry/modules-1/forms-1.xml:20]: is this big label necessary?

File: src/test/resources/archives/case_claim_post_in_entry/modules-1/forms-0.xml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+			<instance src="jr://instance/casedb" id="casedb"/>
+			<instance src="jr://instance/session" id="commcaresession"/>
+			<bind nodeset="/data/welcome_message"/>
+			<bind nodeset="/data/sample_case_logic" type="xsd:string" relevant="instance('casedb')/casedb/case[@case_id = instance('commcaresession')/session/data/case_id]/sample_choice_question = &quot;choice1&quot;"/>
+			<bind nodeset="/data/sample_choice_question"/>
+			<bind nodeset="/data/sample_number_question" type="xsd:int"/>
+			<itext>
+				<translation lang="en" default="">
+					<text id="welcome_message-label">
+						<value>This is a **follow up form**.  It will collect additional or updated information about a previously registered case.  You can then save this information to the case as **case properties**, reference previously collected data, or update properties (ex.  child's weight).
```
Comment (snopoke) [inline src/test/resources/archives/case_claim_post_in_entry/modules-1/forms-0.xml:22]: can we remove this large text blob?

File: src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java (hunk 2 above)
Comment (shubham1g5) [reply src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java:486]: I think that sets the initial itemset choices on the dropdowns based on any defaults (there are test failures after removing that line). I have not looked into it super deeply though.

File: src/test/resources/archives/case_claim_post_in_entry/modules-1/forms-1.xml (hunk 3 above)
Comment (snopoke) [reply src/test/resources/archives/case_claim_post_in_entry/modules-1/forms-1.xml:20]: would still be nice to remove this or shorten it

File: src/test/resources/archives/case_claim_post_in_entry/modules-0/forms-0.xml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+			<bind nodeset="/data/question1/question2" type="xsd:string"/>
+			<bind nodeset="/data/question1/question3" type="xsd:string"/>
+			<bind nodeset="/data/welcome_message"/>
+			<bind nodeset="/data/name" type="xsd:string" required="true()"/>
+			<bind nodeset="/data/sample_choice_question"/>
+			<bind nodeset="/data/sample_number_question" type="xsd:int"/>
+			<itext>
+				<translation lang="en" default="">
+					<text id="welcome_message-label">
+						<value>This is a registration form.  It adds new items (**Cases**) to track.  You can save some or all of the questions in this form to the case as **case properties**, which means you can refer to them in later forms, display them on the phone as a record, or use them to distinguish among cases.
```
Comment (snopoke) [inline src/test/resources/archives/case_claim_post_in_entry/modules-0/forms-0.xml:28]: would still be nice to remove this or shorten it

**Pull request merged**